controllers/datafoundation_client.go

```diff
@@ -0,0 +1,134 @@
+package controllers
+
+import (
+	"fmt"
+
+	v1alpha1 "github.com/red-hat-storage/managed-fusion-agent/api/v1alpha1"
+	"github.com/red-hat-storage/managed-fusion-agent/datafoundationclient"
+	ocsclient "github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+)
+
+type dataFoundationClientSpec struct {
+	onboardingTicket  string
+	storageDFEndpoint string
+}
+
+type dataFoundationClientReconciler struct {
+	*ManagedFusionOfferingReconciler
+
+	offering      *v1alpha1.ManagedFusionOffering
+	spec          dataFoundationClientSpec
+	storageclient ocsclient.StorageClient
+}
+
+//+kubebuilder:rbac:groups=ocs.openshift.io,namespace=system,resources=storageclients,verbs=get;list;watch;create;update;patch;delete
+
+func dfcSetupWatches(controllerBuilder *builder.Builder) {
+	controllerBuilder.
+		Owns(&ocsclient.StorageClient{})
+}
+
+func DFCAddToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(ocsv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(ocsclient.AddToScheme(scheme))
+}
+
+func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	r := dataFoundationClientReconciler{}
+	r.initReconciler(offeringReconciler, offering)
+
+	if err := r.parseSpec(offering); err != nil {
+		return ctrl.Result{}, err
+	}
+	return r.reconcilePhases()
+}
+
+func (r *dataFoundationClientReconciler) initReconciler(reconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) {
+	r.ManagedFusionOfferingReconciler = reconciler
+	r.offering = offering
+}
+
+func (r *dataFoundationClientReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
+	r.Log.Info("Parsing ManagedFusionOffering Data Foundation Client spec")
+
+	isValid := true
+	onboardingTicketAsString, found := offering.Spec.Config["onboarding_ticket"]
+	if !found {
+		r.Log.Error(
+			fmt.Errorf("missing field: onboarding_ticket"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	} else if onboardingTicketAsString == "" {
+		r.Log.Error(
+			fmt.Errorf("onboarding_ticket is empty "),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+
+	storageDFEndpointAsString, found := offering.Spec.Config["storage_DF_Endpoint"]
```

Comment:
storageDFEndpoint is a string unless you want to parse it as an IP. 
Also the key name and variable do not make sense
```suggestion
	dfProviderEndpoint, found := offering.Spec.Config["data_foundation_provider_endpoint"]
```

Reply:
updated